gadopt/stokes_integrators.py

```diff
@@ -228,18 +228,26 @@ def __init__(
         self.mesh = self.Z.mesh()
         self.test = fd.TestFunctions(self.Z)
         self.solution = z
+        self.T = T
         self.approximation = approximation
+        self.quad_degree = quad_degree
 
         self.J = J
         self.constant_jacobian = constant_jacobian
         self.linear = not depends_on(self.approximation.mu, self.solution)
 
         self.solver_kwargs = kwargs
 
+        self.k = upward_normal(self.mesh)
+
         # Setup boundary conditions
         self.weak_bcs = {}
         self.strong_bcs = []
-        free_surface_dict = {}  # Separate dictionary for copying free surface information
+
+        # Free surface parameters
+        self.free_surface_dict = {}  # Separate dictionary for copying free surface information
```

Comment:
What do you mean by "copying free surface information"?

Reply:
sorry that was an unhelpful comment. it is just a dictionary to store user specified parameters about the free surface from a runscript. I've taken the comment out now